implementation/src/main/java/io/smallrye/config/AbstractConfigSource.java

```diff
@@ -0,0 +1,33 @@
+package io.smallrye.config;
+
+import java.io.Serializable;
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+public abstract class AbstractConfigSource implements ConfigSource, Serializable {
+    private int ordinal;
```

Comment:
Making `ordinal` mutable could cause trouble.  It should be final.  That does raise a challenge however for config sources which configure their ordinal from their own data (is that all of them?); it might be worth introducing an intermediate class or classes to support this (e.g. a `MapBasedConfigSource` which accepts a `Map` parameter could call `super(name, Integer.parseInt(map.getOrDefault(...));` and the properties config source could call e.g. `super(propertiesToMap(properties))` or some such).